Fix timelord-install.sh for CentOS\RHEL

LeroyINC opened the pull request:
### Purpose:

This pull request fixes the issue were the install-timelord.sh fails on CentOS and RHEL


### Current Behavior:

When running the install-timelord.sh on CentOS\RHEL the install fails because the name of the python development package that the script tries to install is wrong. For CentOS\RHEL the name is "python3-devel"

If you chose to not to install the python3 development package using the "-n" option  --> "install-timelord.sh -n"
This causes the yum command to have a double space in the command and then yum throws an error and the script aborts.

### New Behavior:

The package name is updated only for CentOS/RHEL, the Package name for other OSes is not changed. The double space issues in the yum command has been fixed.

This results in the expected behavior.

<!-- As we aim for complete code coverage, please include details regarding unit, and regression tests -->
### Testing Notes:

tested by running install-timelord.sh on CentOS 9 system with and without the -n switch

<!-- Attach any visual examples, or supporting evidence (attach any .gif/video/console output below) -->

LeroyINC committed:
Fix timelord-install.sh for CentOS\RHEL

Signed-off-by: LeroyINC <leroy@integrated.net>

Starttoaster commented on install-timelord.sh:
```diff
@@ -102,8 +109,8 @@ else
-		echo "yum install gcc gcc-c++ gmp-devel $PYTHON_DEV_DEPENDENCY libtool make autoconf automake openssl-devel libevent-devel boost-devel python3 cmake -y"
-		sudo yum install gcc gcc-c++ gmp-devel "$PYTHON_DEV_DEPENDENCY" libtool make autoconf automake openssl-devel libevent-devel boost-devel python3 cmake -y
+		echo "yum install gcc gcc-c++ gmp-devel $PYTHON_DEV_DEPENDENCY libtool make autoconf automake openssl-devel libevent-devel boost-devel python3 cmake -y" | tr -s ' '
+		sudo yum install gcc gcc-c++ gmp-devel "$PYTHON_DEV_DEPENDENCY" libtool make autoconf automake openssl-devel libevent-devel boost-devel python3 cmake -y | tr -s ' '
```
I don't have a centos system to test with, but I ran this in a rockylinux container, and I don't think this has the desired effect. It would seem that the `sudo yum` command gets ran as-is before the output of which gets piped to `tr`

Starttoaster replied:
Perhaps something like this would do it?

```bash
yumcmd=$(echo "sudo yum install gcc gcc-c++ gmp-devel $PYTHON_DEV_DEPENDENCY libtool make autoconf automake openssl-devel libevent-devel boost-devel python3 cmake -y" | tr -s ' ')
echo "$yumcmd"
${yumcmd}
```

What do you think @LeroyINC ?

LeroyINC replied:
good catch... and that makes sense.

how about this to make it more simple

if [ "$(uname)" = "Linux" ] && type apt-get; then
	UBUNTU_DEBIAN=true
	echo "Found Ubuntu/Debian."
elif [ "$(uname)" = "Linux" ] && type dnf || yum; then
	RHEL_BASED=true
	echo "Found RedHat."
	
	if [ "$INSTALL_PYTHON_DEV" ]; then
		yumcmd="sudo yum install "$PYTHON_VERSION"-devel gcc gcc-c++ gmp-devel libtool make autoconf automake openssl-devel libevent-devel boost-devel python3 cmake -y"
	else
		yumcmd="sudo yum install gcc gcc-c++ gmp-devel libtool make autoconf automake openssl-devel libevent-devel boost-devel python3 cmake -y"
	fi
	
elif [ "$(uname)" = "Darwin" ]; then
	MACOS=true
	echo "Found MacOS."
fi


then lower down... 

echo "$yumcmd"
${yumcmd}

LeroyINC replied:
updated the code... check it out and let me know

Starttoaster replied:
Thanks for the update! Left one more comment, and I'll resolve this one.

LeroyINC committed:
Update install-timelord.sh


Changed command structure to be more compatible with different flavors of Linux

Starttoaster commented on install-timelord.sh:
```diff
@@ -76,14 +76,23 @@ symlink_vdf_bench() {
 	echo "Found RedHat."
+
+	if [ "$INSTALL_PYTHON_DEV" ]; then
+		yumcmd="sudo yum install "$PYTHON_VERSION"-devel gcc gcc-c++ gmp-devel libtool make autoconf automake openssl-devel libevent-devel boost-devel python3 cmake -y"
```
The use of quotes around `$PYTHON_VERSION` actually unquotes this variable. If you wouldn't mind removing the quotes, I think that's my last piece of feedback before I hit the approve button :) Mostly to appease shellcheck but having it in the quotes is technically correct

LeroyINC replied:
i have fixed this as requested.

Starttoaster replied:
Thank you!

LeroyINC committed:
Update install-timelord.sh


updated as requested.

Starttoaster reviewed (dismissed):
LGTM

Starttoaster commented:
Hi @LeroyINC sorry for the late call out on this. I just noticed because of the CI workflow that there are unsigned commits in this PR. Unfortunately, commit signatures are required for contributions, which can be seen in the contributing.md file/policy here https://github.com/Chia-Network/chia-blockchain/blob/main/CONTRIBUTING.md#creating-signed-commits

LeroyINC commented:
I will see if I can sign this.. not sure how, since it has some other commits\edits after the unsigned commit.

Starttoaster commented:
If git isn't working with you, feel free to close the PR and make a new one from a new branch and just link it to me in here and we can go from there. :)

LeroyINC commented:
ok i have signed my commits...

LeroyINC reviewed (commented):
signed my commits

Starttoaster reviewed (approved):
LGTM
